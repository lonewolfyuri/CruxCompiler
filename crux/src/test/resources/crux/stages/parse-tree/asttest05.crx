// test comparisons

func main() : void {
  var a:float;
  var b:float;

  ::printBool(a < b);
  ::printBool(a <= b);
  ::printBool(a == b);
  ::printBool(a != b);
  ::printBool(a >= b);
  ::printBool(a > b);
}
