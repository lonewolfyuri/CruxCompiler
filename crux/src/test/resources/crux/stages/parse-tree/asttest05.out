program
  declarationList
    declaration
      functionDefinition
        parameterList
        type
        statementBlock
          statementList
            statement
              variableDeclaration
                type
            statement
              variableDeclaration
                type
            statement
              callStatement
                callExpression
                  expressionList
                    expression0
                      expression1
                        expression2
                          expression3
                            designator
                      op0
                      expression1
                        expression2
                          expression3
                            designator
            statement
              callStatement
                callExpression
                  expressionList
                    expression0
                      expression1
                        expression2
                          expression3
                            designator
                      op0
                      expression1
                        expression2
                          expression3
                            designator
            statement
              callStatement
                callExpression
                  expressionList
                    expression0
                      expression1
                        expression2
                          expression3
                            designator
                      op0
                      expression1
                        expression2
                          expression3
                            designator
            statement
              callStatement
                callExpression
                  expressionList
                    expression0
                      expression1
                        expression2
                          expression3
                            designator
                      op0
                      expression1
                        expression2
                          expression3
                            designator
            statement
              callStatement
                callExpression
                  expressionList
                    expression0
                      expression1
                        expression2
                          expression3
                            designator
                      op0
                      expression1
                        expression2
                          expression3
                            designator
            statement
              callStatement
                callExpression
                  expressionList
                    expression0
                      expression1
                        expression2
                          expression3
                            designator
                      op0
                      expression1
                        expression2
                          expression3
                            designator
